server.go

```diff
@@ -33,13 +33,16 @@ func (p *PasswordService) Authenticate(request *Packet) (*Packet, error) {
 }
 
 // NewServer return a new Server given a addr, secret, and service
-func NewServer(addr string, secret string, service Service) *Server {
+func NewServer(addr string, secret string, service Service, clients ...*Clients) *Server {
 	s := &Server{addr: addr,
 		secret:    secret,
 		service:   service,
 		ch:        make(chan struct{}),
 		waitGroup: &sync.WaitGroup{},
 	}
+	if len(clients) > 0 && clients[0] != nil {
```

Comment:
This is also confused me.
What if I pass two clients into it?
Why does it only use the first one.
A new function with a NewServerRequest struct can make all those complexity for caller disappear.